clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -1399,6 +1404,184 @@ public void testInvalidTopicNames() throws Exception {
         }
     }
 
+    @SuppressWarnings("NPathComplexity")
```

Comment:
Could we have a test with an invalid cursor?

Reply:
Do you mean the server returns an invalid cursor or the client sends one?

Reply:
The client sends one. There will be multiple client implementations over time, so it's prudent to ensure that a malformed cursor is caught properly, just in case.

Reply:
Updated the server-side code and added more tests with invalid cursor.